source/guides/recipes/autoformatting.md

```diff
@@ -0,0 +1,55 @@
+(autoformatting)=
+# Autoformatting
+
+[`nixfmt`](https://github.com/NixOS/nixfmt) is the official Nix autoformatter.
+It is not yet wired up to any official tooling. Subscribe to
+[NixOS/nix PR #11252](https://github.com/NixOS/nix/pull/11252) for updates on that effort.
+
+Because `nixfmt` doesn't support formatting whole directory trees, you need
+additional tooling such as `treefmt`. The `nixfmt-tree` package provides a
+`treefmt` pre-configured to run `nixfmt` on all nix files in your project. Just
+add it to your shell:
+
+```nix
+mkShell {
+  packages = [ pkgs.nixfmt-tree ];
+}
+```
+
+Note: this assumes you're working in a git repository with a [`.git/index` file
+at the
+root](https://github.com/NixOS/nixpkgs/blob/bbcdf758d61b6095693d00dea8a75061bf89b264/pkgs/by-name/ni/nixfmt-tree/package.nix#L33-L34).
+
+If you need to change the root detection, or configure treefmt to format other (non-nix)
```

Comment:
Maybe better to be more general and less specific regarding treefmt options?

```suggestion
If you need to configure any [treefmt options], or enable formatting other (non-nix)
```

We can also use this as an opportunity to link to the treefmt config docs:

```md
[treefmt options]: https://treefmt.com/latest/getting-started/configure/#global-options
```